certificate.go

```diff
@@ -93,24 +93,23 @@ func (c Certificate) Expires() time.Time {
 
 // GetFingerprints returns the list of certificate fingerprints, one of which
 // is computed with the digest algorithm used in the certificate signature.
-func (c Certificate) GetFingerprints() []DTLSFingerprint {
+func (c Certificate) GetFingerprints() ([]DTLSFingerprint, error) {
 	fingerprintAlgorithms := []dtls.HashAlgorithm{dtls.HashAlgorithmSHA256}
 	res := make([]DTLSFingerprint, len(fingerprintAlgorithms))
 
 	i := 0
 	for _, algo := range fingerprintAlgorithms {
 		value, err := dtls.Fingerprint(c.x509Cert, algo)
 		if err != nil {
-			fmt.Printf("Failed to create fingerprint: %v\n", err)
-			continue
+			return nil, fmt.Errorf("failed to create fingerprint: %v", err)
```

Comment:
Doesn't this mean the function will fail if any of the fingerprints fail rather than choosing the last working fingerprint like before?